OpenRA.Mods.Common/Widgets/Logic/Editor/MapEditorLogic.cs

```diff
@@ -20,10 +22,40 @@ public class MapEditorLogic : ChromeLogic
 	{
 		MapCopyFilters copyFilters = MapCopyFilters.All;
 
+		[TranslationReference]
+		static readonly string SaveCurrentMap = "save-current-map";
+
 		[ObjectCreator.UseCtor]
-		public MapEditorLogic(Widget widget, World world, WorldRenderer worldRenderer)
+		public MapEditorLogic(Widget widget, World world, WorldRenderer worldRenderer, ModData modData)
 		{
 			var editorViewport = widget.Get<EditorViewportControllerWidget>("MAP_EDITOR");
+			var actionManager = world.WorldActor.Trait<EditorActionManager>();
+
+			var saveButton = widget.GetOrNull<ButtonWidget>("SAVE_BUTTON");
```

Comment:
Tbh I don't like this approach, we should be reusing the same logic from #20123. The button also shows up only in TD, in other mods it's invisible

it is also in an awkward place, shouldn't it be next to undo?
<img width="695" alt="Screenshot 2022-08-30 at 12 52 29" src="https://user-images.githubusercontent.com/37534529/187407976-32237ecb-5193-447b-a110-48f71d2535c1.png">

Reply:
Another question, do we need the button at all? The user already has access to it vie menu. And even if we decide that it is needed it should be renamed to `Quick Save`

Reply:
We should also consider adding a keybind the the ingame-menu save. It should be `ctrl + shift + S`, as in `Save As`

Reply:
the the other button could also remain named as `Save`

Reply:
I think for this PR it's enough to just add the quick-save button and simply label it "Save". The tooltip already states that this is a "quick save".

![image](https://user-images.githubusercontent.com/1355810/187414850-1a0a7e06-117a-4338-bb7e-6c59538faad0.png)

Reply:
Maybe, but we should plan where these buttons need to be longterm, otherwise the next PR is just going to revert this one

Reply:
Yeah, the positioning is clearly off. I think it belongs next to undo and redo - as these are all actions. A major redesign of the Editor UI is brewing up (see #20226) so I don't we should spend much time with that here.

Reply:
well I think we should at the minimum consider `save as`

Reply:
Removed the button.